### PR TITLE
Partial implemention of Vector Search

### PR DESCRIPTION
**Pull request by bmeike**
Includes some tests and a previous commit that fixes some old tests

This is work in progress.  Many of the methods are incomplete.

### CODE REVIEW

File: android/main/java/com/couchbase/lite/CouchbaseLite.java [hunk 1]
```diff
@@ -24,49 +24,98 @@
...
+     * Initialize CouchbaseLite library. Call this method before using CouchbaseLite.
      *
+     * @param ctxt  the ApplicationContext.
      * @param debug true to enable debugging
-     * @throws IllegalStateException on initialization failure
      */
     public static void init(@NonNull Context ctxt, boolean debug) {
-        init(ctxt, debug, ctxt.getFilesDir(), new File(ctxt.getFilesDir(), CouchbaseLiteInternal.SCRATCH_DIR_NAME));
+        final File filesDir = ctxt.getFilesDir();
+        init(ctxt, DEFAULT_EXT_DIR, debug, filesDir, new File(filesDir, CouchbaseLiteInternal.SCRATCH_DIR_NAME));
```
Comment (borrrden) [inline android/main/java/com/couchbase/lite/CouchbaseLite.java:57]: Doesn't this bring back the issue yesterday of "there is no way to tell if the developer meant to use the extension directory or not"?

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java [hunk 2]
```diff
@@ -60,6 +60,7 @@ private CouchbaseLiteInternal() { }
     public static final String PLATFORM = "CBL-ANDROID";
 
     public static final String SCRATCH_DIR_NAME = "CouchbaseLiteTemp";
+    public static final String VS_EXTENSION_NAME = "CouchbaseLiteVectorSearch.so";
```
Comment (borrrden) [inline android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java:63]: Probably not going to be used?

File: common/main/cpp/com_couchbase_lite_internal_core_impl_NativeC4.h [hunk 3]
```diff
@@ -46,6 +46,14 @@ JNIEXPORT void JNICALL Java_com_couchbase_lite_internal_core_impl_NativeC4_debug
 JNIEXPORT void JNICALL Java_com_couchbase_lite_internal_core_impl_NativeC4_setTempDir
         (JNIEnv *, jclass, jstring);
 
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4
+ * Method:    setTempDir
```
Comment (borrrden) [inline common/main/cpp/com_couchbase_lite_internal_core_impl_NativeC4.h:51]: Invalid comment

File: common/main/cpp/native_c4.cc [hunk 4]
```diff
@@ -234,6 +236,17 @@ Java_com_couchbase_lite_internal_core_impl_NativeC4_setTempDir(JNIEnv *env, jcla
         throwError(env, error);
 }
 
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4
+ * Method:    setTempDir
```
Comment (borrrden) [inline common/main/cpp/native_c4.cc:241]: Another invalid comment

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java [hunk 5]
```diff
@@ -168,6 +169,10 @@ public static Map<String, String> loadErrorMessages(@NonNull Context ctxt) {
         return errorMessages;
     }
 
+    @VisibleForTesting
+    public static void setExtensionPath(@NonNull Context ctxt, @NonNull File extDir, @NonNull File scratchDir) {
```
Comment (borrrden) [inline android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java:173]: What is the scratchDir for here?

**Review by borrrden — COMMENTED** [review]
Ended up with more comments than I thought

File: android/main/java/com/couchbase/lite/CouchbaseLite.java (hunk 1 above)
Comment (bmeike) [reply android/main/java/com/couchbase/lite/CouchbaseLite.java:57]: no....   client code can't explicitly specify default.   I need to check that.

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java (hunk 2 above)
Comment (bmeike) [reply android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java:63]: maybe not, now.   It was when I was copying it out of assets.

File: common/main/cpp/com_couchbase_lite_internal_core_impl_NativeC4.h (hunk 3 above)
Comment (bmeike) [reply common/main/cpp/com_couchbase_lite_internal_core_impl_NativeC4.h:51]: yep.  fixed.

File: common/main/cpp/native_c4.cc (hunk 4 above)
Comment (bmeike) [reply common/main/cpp/native_c4.cc:241]: Yep.  fixed

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java (hunk 5 above)
Comment (bmeike) [reply android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java:173]: It is where the asset used to be copied.

**Review by pasin — COMMENTED** [review]
Do we release the vector search extension for android as an AAR on a maven server or a downloadable zip file containing .so files?

If we are doing AAR on a maven server, I understand that users will use the vector search like below.

```
implementation 'com.couchbase.lite:couchbase-lite-android-ee:3.2.0'
implementation 'com.couchbase.lite:couchbase-lite-vector-search-extension:1.0.0'
```

This would mean that the location of the vector search binary will be where the Android OS put the Native libraries for the application. So I'm not sure if there will be cases that user will customize the location beside using the default location.

**Review by pasin — APPROVED** [review]

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java (hunk 2 above)
Comment (bmeike) [reply android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java]: Definitely used, now

File: android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java (hunk 5 above)
Comment (bmeike) [reply android/main/java/com/couchbase/lite/internal/CouchbaseLiteInternal.java]: Removed

**Comment by bmeike** [comment]
Changed to remove client specified extension path

**Review by borrrden — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Pull request merged**